### ISSUE TITLE
Events added to in-memory queue are not being raised after ContinueAsNew called.

### ISSUE THREAD

**Issue by revoxover** [Azure/azure-functions-durable-extension#676]
Perhaps I misunderstood the functionality added in version 1.8.0 to address the "Race conditions causing external events to be dropped" issues but I expected the following scenario to work.

I have a Orchestration running with the following code:
```
[FunctionName("RunOrchestrator")]
public static async Task RunOrchestrator(
	[OrchestrationTrigger] DurableOrchestrationContext context,
	ILogger log)
{

	//wait for event
	log.LogWarning($"Waiting for event.");
	var raisedEvent = await context.WaitForExternalEvent<string>("Event");

	//handle event
	log.LogWarning($"Event caught: {raisedEvent}");
	await context.CallActivityAsync("HandleEvent", raisedEvent);

	//call continue
	log.LogWarning($"Calling ContinueAsNew().");
	context.ContinueAsNew(null);
}
```

In the "HandleEvent" activity I have an **await Task.Delay(10000);** to simulate some work being done. When I raise events in quick succession only the first event is handled and the others are added to an in memory queue as expected with the following output message:

_"Function 'RunOrchestrator (Orchestrator)' saved a 'Event' event to an in-memory queue. State: ExternalEventDropped."_

However once the first handled event completes and **ContinueAsNew()** is called the events that are in the queue are not being raised once the **context.WaitForExternalEvent** line is hit again.

Is this scenario supposed to work? Thanks in advance.

**Comment by cgillum** [comment]
Yes, this scenario is supposed to work. We have a test intended to validate it [here](https://github.com/Azure/azure-functions-durable-extension/blob/cf6b0b37ad646fe5c65d0965653b0cb136763174/test/Common/DurableTaskEndToEndTests.cs#L627).

I'll try modifying our test to better match yours and see if I can reproduce the issue.

**Comment by cgillum** [comment]
I confirmed that adding the delay like you mentioned seems to cause an issue. I'll investigate further and see if we can make a fix.

**Comment by revoxover** [comment]
Thanks you, much appreciated.

**Comment by cgillum** [comment]
This fix has been released in the [v1.8.1 release](https://github.com/Azure/azure-functions-durable-extension/releases/tag/v1.8.1).
